### PR TITLE
allow emails to be passed through SAML

### PR DESCRIPTION
**Pull request by cg505**
Signed-off-by: Christopher Cooper <cooperc@ocf.berkeley.edu>

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

### CODE REVIEW

**Comment by ptman** [comment]
A sample config would be welcome as well. I tried to provide something at https://github.com/matrix-org/synapse/issues/5764#issuecomment-625194187

**Comment by clokep** [comment]
I caused a conflict with this branch in #7458, sorry about that. Could you merge in develop into this branch?

**Comment by cg505** [comment]
@ptman The config hasn't changed at all in this PR, so I don't know that this is the right place.

@clokep Thank you for the heads up, I will take care of it when I get a chance (probably this weekend).

**Comment by ptman** [comment]
@cg505 ok, so once this is merged, emails are automatically picked up from saml (as long as the attirbute name is 'email')?

File: changelog.d/7385.feature [hunk 1]
```diff
@@ -0,0 +1 @@
+Add the ability to pass an email via SAML attributes.
```
Comment (richvdh) [inline changelog.d/7385.feature:1]:
```suggestion
For SAML authentication, add the ability to pass email addresses to be added to new users' accounts via SAML attributes. Contributed by Christopher Cooper.
```

**Comment by richvdh** [comment]
looks like this fixes #7023

**Comment by cg505** [comment]
@anoadragon453 Thanks for taking care of it.

**Comment by ptman** [comment]
And just to check. This only applies to new users, does not update existing users on new login?

**Comment by cg505** [comment]
believe so

**Pull request merged**